src/map/basic_map.rs

```diff
@@ -110,6 +117,43 @@ impl Map {
     self.stairs_up = Some(Point::new(stairs_position.0, stairs_position.1));
   }
 
+  pub fn add_doors_to_rooms(&mut self) {
+    for i in 0..self.rooms.len() {
```

Comment:
you could use for `room in self.rooms` here to avoid indexing into self.rooms on lines 122, 123, 126 and 140

Reply:
That's great -- that's how I would definitely prefer to do it I just wasn't sure the syntax 👍